integration-tests/tests/singlestore/singlestore-common.ts

```diff
@@ -0,0 +1,3440 @@
+				.orderBy(asc(users.id))
+				.prepare();
+			const iter = prepared.iterator();
+			const result: typeof users.$inferSelect[] = [];
+
+			for await (const row of iter) {
+				result.push(row);
+			}
+
+			expect(result).toEqual([{ id: 1 }, { id: 2 }, { id: 3 }]);
+		});
+
+		test('insert undefined', async (ctx) => {
+			const { db } = ctx.singlestore;
+
+			const users = singlestoreTable('users', {
+				id: serial('id').primaryKey(),
+				name: text('name'),
+			});
+
+			await db.execute(sql`drop table if exists ${users}`);
+
+			await db.execute(
+				sql`create table ${users} (id serial not null primary key, name text)`,
+			);
+
+			await expect((async () => {
+				await db.insert(users).values({ name: undefined });
+			})()).resolves.not.toThrowError();
+
+			await db.execute(sql`drop table ${users}`);
+		});
+
+		test('update undefined', async (ctx) => {
+			const { db } = ctx.singlestore;
+
+			const users = singlestoreTable('users', {
+				id: serial('id').primaryKey(),
+				name: text('name'),
+			});
+
+			await db.execute(sql`drop table if exists ${users}`);
+
+			await db.execute(
+				sql`create table ${users} (id serial not null primary key, name text)`,
+			);
+
+			await expect((async () => {
+				await db.update(users).set({ name: undefined });
+			})()).rejects.toThrowError();
+
+			await expect((async () => {
+				await db.update(users).set({ id: 1, name: undefined });
+			})()).resolves.not.toThrowError();
+
+			await db.execute(sql`drop table ${users}`);
+		});
+
+		test('utc config for datetime', async (ctx) => {
+			const { db } = ctx.singlestore;
+
+			await db.execute(sql`drop table if exists \`datestable\``);
+			await db.execute(
+				sql`
+					create table \`datestable\` (
+					    \`datetime_utc\` datetime(6),
+					    \`datetime\` datetime(6)
+					)
+				`,
+			);
+			const datesTable = singlestoreTable('datestable', {
+				datetimeUTC: datetime('datetime_utc', { fsp: 6, mode: 'date' }),
+				datetime: datetime('datetime', { fsp: 6 }),
+			});
+
+			const dateObj = new Date('2022-11-11');
+			const dateUtc = new Date('2022-11-11T12:12:12.122Z');
+
+			await db.insert(datesTable).values({
+				datetimeUTC: dateUtc,
+				datetime: dateObj,
+			});
+
+			const res = await db.select().from(datesTable);
+
+			const [rawSelect] = await db.execute(sql`select \`datetime_utc\` from \`datestable\``);
+			const selectedRow = (rawSelect as unknown as [{ datetime_utc: string }])[0];
+
+			expect(selectedRow.datetime_utc).toBe('2022-11-11 12:12:12.122000');
+			expect(new Date(selectedRow.datetime_utc.replace(' ', 'T') + 'Z')).toEqual(dateUtc);
+
+			expect(res[0]?.datetime).toBeInstanceOf(Date);
+			expect(res[0]?.datetimeUTC).toBeInstanceOf(Date);
+
+			expect(res).toEqual([{
+				datetimeUTC: dateUtc,
+				datetime: new Date('2022-11-11'),
+			}]);
+
+			await db.execute(sql`drop table if exists \`datestable\``);
+		});
+
+		test('set operations (union) from query builder with subquery', async (ctx) => {
```

Comment:
just a note to flag this for review since it differs from the MySQL tests, and is technically "broken" on the implementation side

Reply:
Could you clarify what you mean by "broken" and "implementation side" please?

Reply:
"broken" meaning that the `orderBy` and `limit` cannot be chained right now, so this test had to be modified to work around it for now (MCDB-63261)